src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -976,20 +976,10 @@ internal static string GetPSHomeModulePath()
             try
             {
                 string psHome = Utils.DefaultPowerShellAppBase;
-                if (!string.IsNullOrEmpty(psHome))
-                {
-                    // Win8: 584267 Powershell Modules are listed twice in x86, and cannot be removed
-                    // This happens because ModuleTable uses Path as the key and CBS installer
-                    // expands the path to include "SysWOW64" (for
-                    // HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Microsoft\PowerShell\3\PowerShellEngine ApplicationBase).
-                    // Because of this, the module that is getting loaded during startup (through LocalRunspace)
-                    // is using "SysWow64" in the key. Later, when Import-Module is called, it loads the
-                    // module using ""System32" in the key.
 #if !UNIX
-                    psHome = psHome.ToLowerInvariant().Replace("\\syswow64\\", "\\system32\\");
+                psHome = psHome.ToLowerInvariant().Replace(@"\syswow64\", @"\system32\");
```

Comment:
Well, like it or not, a user can install x86 PowerShell under `C:\Windows\SysWOW64` 🤷 

![image](https://user-images.githubusercontent.com/127450/148276871-b0bf0b56-ba1d-4b5a-99b9-538f86a4b557.png)

Reply:
User can. But in the case the path will be a regular path and the code is not for the scenario (see comment in code).
I think we should remove the code since it misleads.

Reply:
I don't think I really understand that comment, especially the `the module that is getting loaded during startup (through LocalRunspace)` part. I'm not sure how modules are loaded that.

But one thing that is sure is for Windows PowerShell x86, the `$PSHOME` is `C:\Windows\SysWOW64\WindowsPowerShell\v1.0`, so if the `SysWOW64` cause some modules in `$PSHOME\Modules` to be loaded twice as different module due to different keys (path) in `ModuleTable`, it could happen to PowerShell Core x86 as well when it's installed under `C:\Windows\SysWOW64`.

Unless you can repro the `listed twice` issue, and then demonstrate it's not a problem in PowerShell Core without this line of code, I'm afraid removing this line of code may break some scenario.
Actually, I should keep that comment around.

Reply:
> it could happen to PowerShell Core x86 as well when it's installed under C:\Windows\SysWOW64.

1. We never install pwsh in the folder with msi. So the code makes no sense for pwsh.
2. Since we install pwsh with msi in `c:\Program Files x86` the double loading problem could be raised in the "mirror" case but the code doesn't address this. This is why I think we should base it on the architecture of the pwsh process and maybe take into account the architecture of the loadable module.

Reply:
> We never install pwsh in the folder with msi. So the code makes no sense for pwsh.

User controls where to install pwsh, so it's possible, see my screenshot above.

> we install pwsh with msi in c:\Program Files x86 the double loading problem could be **raised in the "mirror" case** but the code doesn't address this.

I'm not sure what you mean by the bold part. But nevertheless, this is out of the scope of this PR.